src/openforms/forms/tests/admin/test_tasks.py

```diff
@@ -122,3 +127,115 @@ def test_import_form_failure(self, m_import_form):
 
         for item in failed_forms:
             self.assertEqual(item[1], ["Something went wrong"])
+
+
+class ActivateFormsTests(TestCase):
+    @freeze_time("2023-10-10T21:15:00Z")
+    def test_forms_are_activated_on_specified_datetime(self):
+        form1 = FormFactory(active=False, activate_on="2023-10-10T21:15:00Z")
+        form2 = FormFactory(active=False, activate_on="2023-10-10T21:15:00Z")
+
+        activate_forms()
+
+        form1.refresh_from_db()
+        form2.refresh_from_db()
+
+        self.assertTrue(form1.active)
+        self.assertIsNone(form1.activate_on)
+        self.assertTrue(form2.active)
+        self.assertIsNone(form2.activate_on)
+
+    @freeze_time("2023-10-10T21:15:00Z")
+    def test_form_activation_is_logged(self):
+        form = FormFactory(active=False, activate_on="2023-10-10T21:15:00Z")
+
+        activate_forms()
+
+        log_entry = TimelineLogProxy.objects.last()
+        message = log_entry.get_message()
+
+        self.assertEqual(log_entry.content_object.id, form.id)
+        self.assertEqual(
+            message.strip().replace("&quot;", '"'),
+            f"{log_entry.fmt_lead}: Form was activated.",
+        )
+
+    @freeze_time("2023-10-10T21:15:00Z")
+    def test_form_is_not_activated_on_different_time(self):
+        form = FormFactory(active=False, activate_on="2023-10-10T21:16:00Z")
+
+        activate_forms()
+
+        form.refresh_from_db()
+
+        self.assertFalse(form.active)
+        self.assertIsNotNone(form.activate_on)
+
+    @freeze_time("2023-10-10T21:15:00Z")
+    def test_form_is_not_activated_when_soft_deleted(self):
+        form = FormFactory(active=False, activate_on="2023-10-10T21:15:00Z")
+        form._is_deleted = True
+        form.save(update_fields=["_is_deleted"])
+
+        activate_forms()
+
+        form.refresh_from_db()
+
+        self.assertFalse(form.active)
+        self.assertIsNotNone(form.activate_on)
+
```

Comment:
I think you also need a test where you have a form that should be activated at `2023-10-11T21:15:00Z` (not the same date, but the same minute) and make sure that at `2023-10-10T21:15:00Z` it is not activated.